### PR TITLE
More author api

### PR DESCRIPTION
**Pull request by sandysaji**
This PR request includes:
- Checking if two users are friends or not. ('/author/{}/friends/{}')
- Updated all the test cases in test_author_api.py to check for multiple scenarios including user not found.

Changes after review:
- Moves the check_friends() to friends_views.py. 
- Updated the test case as well 

Please let me know if it requires any changes!

### CODE REVIEW

**Review by roychowd — DISMISSED** [review]

File: backend/tests/test_author_api.py [hunk 1]
```diff
@@ -43,3 +66,56 @@ def test_get_friends(self, client, test_user, friend_user):
         assert response.data["Author"] is not None
         assert response.data["Author"][0] == friend_user[0].get_full_user_id()
         assert response.data["Author"][1] == friend_user[1].get_full_user_id()
+
+        '''
+            Checking scenario where the user does not exist, 404 Error
+        '''
```
Comment (maharshmellow) [inline backend/tests/test_author_api.py]: Can we get rid of all these block comments. They should only be used as a doc string and not as a regular comment. Just convert it to a regular comment.

**Review by maharshmellow — CHANGES_REQUESTED** [review]

File: backend/tests/test_author_api.py (hunk 1 above)
Comment (maharshmellow) [reply backend/tests/test_author_api.py]: There are a few in this file and the other PR that got merged as well.

File: backend/apiviews/author_views.py [hunk 2]
```diff
@@ -60,3 +60,24 @@ def get_friends(self, request, pk, *args, **kwargs):
             id_List.append(list(list(i.items())[0][1].items())[0][1])
 
         return Response({"query": "friends", "Author": id_List})
+    
+    def check_friends(self, request, authorId1,authorId2, *args, **kwargs):
+    
+        id1 = protocol_removed(authorId1)
```
Comment (maharshmellow) [inline backend/apiviews/author_views.py]: Can we just change this function name to something like `remove_protocol` instead.

File: backend/apiviews/author_views.py [hunk 3]
```diff
@@ -60,3 +60,24 @@ def get_friends(self, request, pk, *args, **kwargs):
...
+        id2 = protocol_removed(authorId2)
+        author1 = get_object_or_404(User, fullId=id1)
+        author2 = get_object_or_404(User, fullId=id2)
+        friend_ids = Friend.objects.filter(
+            fromUser_id=author1.id).values_list('toUser_id', flat=True)
+        friends_list = []
+        for id in friend_ids:
+            friends_list.append(id)
```
Comment (maharshmellow) [inline backend/apiviews/author_views.py]: What is this function doing? It seems to be just copying items from friends_ids to friends_list so why not just use friend_ids?

File: backend/tests/test_author_api.py [hunk 4]
```diff
@@ -43,3 +66,56 @@ def test_get_friends(self, client, test_user, friend_user):
...
+        assert secondResponse.data["authors"] == [test_user.get_full_user_id(),friend_user[1].get_full_user_id()]
+        assert secondResponse.data['friends'] == False
+
+        '''
+        Checking scenario where the user does not exist, 404 Error
+        '''
+        nouserResponse = client.get('/author/{}/friends/{}'.format(test_auth_id,"https://cmput404-socialdistribution.herokuapp.com/author/20"))
+        assert nouserResponse.status_code == 404
+
+
```
Comment (maharshmellow) [inline backend/tests/test_author_api.py]: get rid of these empty lines

File: backend/urls.py [hunk 5]
```diff
@@ -61,4 +61,7 @@
     path('author/<path:pk>/friends', AuthorViewSet.as_view(({
         "get": "get_friends"
     }))),
+    path('author/<path:authorId1>/friends/<path:authorId2>',AuthorViewSet.as_view(({
+        "get": "check_friends"
+        }))),
```
Comment (maharshmellow) [inline backend/urls.py]: indent this line back to match line 64

**Review by maharshmellow — CHANGES_REQUESTED** [review]

File: backend/tests/test_author_api.py [hunk 6]
```diff
@@ -8,6 +8,15 @@
 class TestAuthorAPI:
 
     def test_get_profile_by_author_id(self, client, test_user, friend_user):
+        '''
+            Checking scenario where the user does not exist, 404 Error
+        '''
+        nouserResponse = client.get('/author/{}/'.format("https://cmput404-socialdistribution.herokuapp.com/author/20"))
```
Comment (zhaoyuxuan) [inline backend/tests/test_author_api.py]: Change this id to something more “wrong" in case the user actually exists in the db

**Review by zhaoyuxuan — CHANGES_REQUESTED** [review]

File: backend/tests/test_author_api.py (hunk 1 above)
Comment (roychowd) [reply backend/tests/test_author_api.py]: Sure.

Comment (sandysaji) [reply backend/tests/test_author_api.py]: Done!

File: backend/apiviews/author_views.py (hunk 3 above)
Comment (sandysaji) [reply backend/apiviews/author_views.py]: This function is being called in other API views too so we might have to change all of those as well

File: backend/tests/test_author_api.py (hunk 4 above)
Comment (sandysaji) [reply backend/tests/test_author_api.py]: Done!

File: backend/urls.py (hunk 5 above)
Comment (sandysaji) [reply backend/urls.py]: Done!

File: backend/tests/test_author_api.py (hunk 6 above)
Comment (sandysaji) [reply backend/tests/test_author_api.py]: Done!

File: backend/apiviews/author_views.py (hunk 3 above)
Comment (sandysaji) [reply backend/apiviews/author_views.py]: I updated this. friends_id returned a QuerySet and now i am just checking if the ids exist in there

File: backend/apiviews/author_views.py (hunk 2 above)
Comment (sandysaji) [reply backend/apiviews/author_views.py]: I believe this function is being called in other views too (not sure which ones). So we would have to replace those ones too

**Review by zhaoyuxuan — APPROVED** [review]

**Review by roychowd — APPROVED** [review]
Thanks for changing FriendViewSet!

File: backend/apiviews/author_views.py (hunk 2 above)
Comment (maharshmellow) [reply backend/apiviews/author_views.py]: sure I can change it in another PR then

**Review by maharshmellow — APPROVED** [review]

**Pull request merged**